Resolver/FormAnnotationResolver.php

```diff
@@ -159,12 +159,12 @@ protected function getBuiltObject(
         $entity = $request->attributes->get($annotation->getEntity());
 
         /**
-         * Creates form object from type
+         * Creates form object from type.
          */
-        $form = $formFactory->create($type, $entity);
+        $form = $formFactory->create(get_class($type), $entity);
```

Comment:
i guess you mean "what", not "why" ;) 
I just checked the code, you are right, this is correct :)